src/cloudant/database.py

```diff
@@ -1098,3 +1101,128 @@ def get_query_result(self, selector, fields=None, raw_result=False,
+            limits the number of documents per group.
+        :param query: A Lucene query in the form of ``name:value``.
+            If name is omitted, the special value ``default`` is used.
+        :param ranges: Optional JSON facet syntax that reuses the standard
+            Lucene syntax to return counts of results which fit into each
+            specified category. Inclusive range queries are denoted by brackets.
+            Exclusive range queries are denoted by curly brackets.
+            For example ``ranges={"price":{"cheap":"[0 TO 100]"}}`` has an
+            inclusive range of 0 to 100.
+            Requires the index to have faceting enabled.
+        :param sort: Optional JSON string of the form ``fieldname<type>`` for
+            ascending or ``-fieldname<type>`` for descending sort order.
+            Fieldname is the name of a string or number field and type is either
+            number or string or a JSON array of such strings. The type part is
+            optional and defaults to number.
+        :param str stale: Optional string to allow the results from a stale
+            index to be used. This makes the request return immediately, even
+            if the index has not been completely built yet.
+        :param list highlight_fields: Optional list of fields which should be
+            highlighted.
+        :param str highlight_pre_tag: Optional string inserted before the
+            highlighted word in the highlights output.  Defaults to ``<em>``.
+        :param str highlight_post_tag: Optional string inserted after the
+            highlighted word in the highlights output.  Defaults to ``<em>``.
+        :param int highlight_number: Optional number of fragments returned in
+            highlights. If the search term occurs less often than the number of
+            fragments specified, longer fragments are returned.  Default is 1.
+        :param int highlight_size: Optional number of characters in each
+            fragment for highlights.  Defaults to 100 characters.
+        :param list include_fields: Optional list of field names to include in
+            search results. Any fields included must have been indexed with the
+            ``store:true`` option.
+
+        :returns: Search query result data in JSON format
+        """
+        if not query_params.get('query'):
+            raise CloudantArgumentError(
+                'No query parameter found.  Please add a query parameter '
+                'containing Lucene syntax and retry.')
+
+        # Validate query arguments and values
+        for key, val in iteritems_(query_params):
+            if key not in list(SEARCH_INDEX_ARGS.keys()):
+                msg = 'Invalid argument: {0}'.format(key)
+                raise CloudantArgumentError(msg)
+            if not isinstance(val, SEARCH_INDEX_ARGS[key]):
+                msg = (
+                    'Argument {0} is not an instance of expected type: {1}'
```

Comment:
Looks like it could be more common for a user to see the singular case knowing that 6 of the 18 query parameters accept multiple types.  If we change this instance of the message to `an instance of one of the expected types`, we would also want to consider changing the other instances in the project.